Push changes to dummy-proof tropo options

sssangha opened the pull request:
Access test products for this case example as so:
```
wget https://gunw-development-testing.s3.us-west-2.amazonaws.com/unavco_2024/S1-GUNW-A-R-124-tops-20180502_20180408-043026-00157W_00018N-PP-9909-v3_0_1.nc .
wget https://gunw-development-testing.s3.us-west-2.amazonaws.com/unavco_2024/S1-GUNW-A-R-124-tops-20180502_20180408-043052-00157W_00019N-PP-7296-v3_0_1.nc .
```

If a user specifies the extraction of a troposphere layer, but neglects to specify an argument for `--tropo_models`, error out early instead of silently failing to extract it.

To test this, run the following command:
```
ariaExtract.py -f "*.nc" -w tropo_test1 -d Download -l troposphereTotal
```

If a user specifies a tropo model, but neglects to specify the extraction of a valid troposphere layer with the '-l' option, error out early instead of silently failing to extract it.

To test this, run the following command:
```
ariaExtract.py -f "*.nc" -w tropo_test2 -d Download --tropo_models ERA5
```

Finally, if a user specifies a tropo model that doesn't exist, catch this early and error out

To test this, run the following command:
```
ariaExtract.py -f "*.nc" -w tropo_test3 -d Download --tropo_models FAKE_MODEL -l troposphereTotal
```

Commit:
Push changes to dummy-proof tropo inputs

Commit:
Add early check to vet if model name is valid

sssangha commented:
@alexfore , I tried to simplify the logic for this dummy-proofing the best I can in this first draft, so please advise if there is any way we should streamline or improve on this.

Also, can you please confirm whether the imports for the [product.py ](https://github.com/aria-tools/ARIA-tools/blob/3cd16d6086e5188dcb5f42cc6fa18d1780eccfb3/tools/ARIAtools/product.py)script are organized appropriately? Thanks

alexfore commented:
I think my only comment is that the copy does not seem needed as VALID_TROPO_MODELS is not being modified but fine as is

alexfore commented on tools/ARIAtools/product.py:
```diff
@@ -714,6 +722,49 @@ def __mappingVersion__(self, fname, version):
 
+                # exit if user does not specify a valid tropo model name
+                if self.tropo_models is not None:
+                    if self.tropo_models.lower() != 'all':
```
Sorry, I found an issue with this line when running the regression tests. I found that `self.tropo_models` may be a list, in which case the `self.tropo_models.lower()` raises `AttributeError`. Wrapping this block in
```
                    with contextlib.suppress(AttributeError):
```
avoids that issue, or a similar solution will work

sssangha replied:
Hello @alexfore , I went ahead and adjusted the logic. Before my tweaks, the user input for `tropo_model` was tested for every file in a loop. This is only necessary to check once, and for only Sentinel-1 products (it is not currently relevant for NISAR products). Please let me know what you think. Thanks

alexfore replied:
OK, it passes the tests now

sssangha committed:
Avoid redundant tropo_model check

Before this commit, the user input for tropo_model was tested for every file in a loop. This is only necessary to check once, and for only Sentinel-1 products (it is not currently relevant for NISAR products)

pep8speaks commented:
Hello @sssangha! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:







There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2024-07-24 22:56:56 UTC

sssangha committed:
pep8 tweaks for tropo_model error log

sssangha committed:
pep8 tweak to tropo_model check attempt 2
